test/src/unit-cppapi-max-fragment-size.cc

```diff
@@ -503,3 +511,383 @@ TEST_CASE(
 
   array.close();
 }
+
+std::optional<uint64_t> subarray_num_cells(
+    std::span<const templates::Domain<uint64_t>> subarray) {
+  uint64_t num_cells = 1;
+  for (const auto& dim : subarray) {
+    auto maybe = checked_arithmetic<uint64_t>::mul(num_cells, dim.num_cells());
+    if (!maybe.has_value()) {
+      return std::nullopt;
+    }
+    num_cells = maybe.value();
+  }
+  return num_cells;
+}
+
+template <typename Asserter>
+std::vector<std::vector<templates::Domain<uint64_t>>>
+instance_dense_global_order(
+    const Context& ctx,
+    uint64_t max_fragment_size,
+    const std::vector<templates::Dimension<Datatype::UINT64>>& dimensions,
+    const std::vector<templates::Domain<uint64_t>>& subarray) {
+  const std::string array_name = "max_fragment_size_dense_global_order";
+
+  const std::optional<uint64_t> num_cells = subarray_num_cells(subarray);
+  ASSERTER(num_cells.has_value());
+
+  Domain domain(ctx);
+  for (uint64_t d = 0; d < dimensions.size(); d++) {
+    const std::string dname = "d" + std::to_string(d);
+    auto dim = Dimension::create<uint64_t>(
+        ctx,
+        dname,
+        {{dimensions[d].domain.lower_bound, dimensions[d].domain.upper_bound}},
+        dimensions[d].extent);
+    domain.add_dimension(dim);
+  }
+
+  auto a = Attribute::create<int>(ctx, "a");
+  ArraySchema schema(ctx, TILEDB_DENSE);
```

Comment:
Very important that we test row-major and column-major tile orders.  Almost certainly column-major is broken as of this writing.